src/components/Demo.svelte

```diff
@@ -69,65 +70,148 @@ let languages: {
   },
 ] as const;
 
-const urlParams = new URLSearchParams(window.location.search);
-if (urlParams.has("go")) {
-  languageCode.Go = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("go"),
-  );
-}
-if (urlParams.has("c")) {
-  languageCode.C = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("c"),
-  );
-}
-if (urlParams.has("c++")) {
-  languageCode["C++"] = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("c++"),
-  );
-}
-if (urlParams.has("python")) {
-  languageCode.Python = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("python"),
-  );
-}
-if (urlParams.has("typescript")) {
-  languageCode.TypeScript = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("typescript"),
-  );
-}
-if (urlParams.has("tsx")) {
-  languageCode.TSX = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("tsx"),
-  );
-}
+type ShareParameters = {
+  version: number;
+  language: string;
+  code: string;
+  fontSize: string;
+  simplify: boolean;
+  flatSwitch: boolean;
+  highlight: boolean;
+  colors: ColorList;
+};
+
+// ADD-LANGUAGES-HERE
+const languageAliases: Record<string, Language> = {
+  go: "Go",
+  c: "C",
+  cpp: "C++",
+  python: "Python",
+  typescript: "TypeScript",
+  tsx: "TSX",
+};
+
+const reverseLanguageAliases = Object.fromEntries(
+  Object.entries(languageAliases).map(([alias, lang]) => [lang, alias]),
+);
```

Comment:
I am not a fan of this mapping, as if we have multiple aliases for a language - we'll only get the last one into the mapping.
Alternatively, we can write a function to look for the _first_ match (that's usually what I'd expect at a glance) and use that.

That said, it doesn't really matter now - so up to you.